### ISSUE TITLE
WinError 10053 / ""MySQL server has gone away" after reconnecting to server

### ISSUE THREAD

**Issue by mjlm** [datajoint/datajoint-python#292]
I'm using DataJoint in Python 3 inside Jupyter notebooks.

The problem occurs when I work in a notebook and then shut down and re-start my mysql server. After re-starting the server, datajoint fails to re-connect properly:

I can still execute `dj.conn(reset=True)` and get the response 

```
Connecting username@localhost:3306
DataJoint connection (connected) username@localhost:3306
```

but the connection doesn't work: Whenever I try to use datajoint in any way (e.g. create a restriction), I get this error:

```
---------------------------------------------------------------------------
ConnectionAbortedError                    Traceback (most recent call last)
C:\Users\Matthias\Anaconda3\lib\site-packages\pymysql\connections.py in _write_bytes(self, data)
   1003         try:
-> 1004             self._sock.sendall(data)
   1005         except IOError as e:

ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in your host machine

During handling of the above exception, another exception occurred:

[stack trace...]

OperationalError: (2006, "MySQL server has gone away (ConnectionAbortedError(10053, 'An established connection was aborted by the software in your host machine', None, 10053, None))")
```

Perhaps some stale connection is lingering around somewhere? It would be great to get help with this because the only remedy is to re-start the python kernel, which is tedious after setting up a complex workspace.

Thanks,
Matthias

**Comment by eywalker** [comment]
**TL;DR;** call `dj.conn().connect()` to reestablish your connection without restarting your Python session.

Hi @mjlm, thanks for posting this issue. As you discoverd, invoking `dj.conn(reset=True)` actually creates a new connection object that is used by **any future** schemas and tables created with DataJoint, but the new connection does **not** get used by the already existing schemas and tables. I must admit that this is rather confusing and misleading, and we are actively working to revise this interface.

To regain the connection following a database connection interrupt **without** restarting your interactive session, you may simply call `dj.conn().connect()`. This will trigger the already existing database connection object returned by `dj.conn()` to (re)connect to the database as necessary.

In the next minor update to the DataJoint Python, issuing any form of query (e.g. looking at a table content) will automatically attempt a reconnect to database if the connection is broken. This will allow for much more seamless usage of DataJoint in cases like what you described 😄

**Comment by eywalker** [comment]
This issue will be addressed via #279. Closing
